src/pybind/mgr/prometheus/module.py

```diff
@@ -549,23 +550,38 @@ def get_rbd_stats(self):
         # in the object name, the image pool is the pool where the object is
         # located.
 
-        pools = self.get_localized_module_option('rbd_stats_pools', '').split()
-        pools.sort()
+        # Parse rbd_stats_pools option, which is a comma or space separated
+        # list of pool[/namespace] entries. If no namespace is specifed the
+        # stats are collected for every namespace in the pool.
+        pools_string = self.get_localized_module_option('rbd_stats_pools', '')
+        pools = {}
+        for p in re.split('[\s,]', pools_string):
+            s = p.split('/', 2)
```

Comment:
Nit: perhaps add ```if not p: ... continue``` to avoid empty strings caused by "pool1, pool2"-style inputs?

Reply:
Good point. I have fixed this by updating the regex.